default/python/AI/AIDependencies.py

```diff
@@ -797,34 +797,22 @@ def _scale_part_damage(part_damage: Dict[str, int], factor: float) -> Dict[str,
 # </editor-fold>
 
 # <editor-fold desc="Shipyards">
-BLD_SHIPYARD_ORBITAL_DRYDOCK = "BLD_SHIPYARD_ORBITAL_DRYDOCK"
-# ship facilities info, dict keyed by building name, value is (min_aggression, prereq_bldg, base_cost, time)
-# not currently determined dynamically because it is initially used in a location-independent fashion
-# note that BLD_SHIPYARD_BASE is not an absolute prereq for BLD_NEUTRONIUM_FORGE, but is a practical one
 SHIP_FACILITIES = {
-    "BLD_SHIPYARD_BASE": (0, "", 10, 4),
-    "BLD_SHIPYARD_ORBITAL_DRYDOCK": (0, "BLD_SHIPYARD_BASE", 20, 5),
-    "BLD_SHIPYARD_CON_NANOROBO": (fo.aggression.aggressive, "BLD_SHIPYARD_ORBITAL_DRYDOCK", 250, 5),
-    "BLD_SHIPYARD_CON_GEOINT": (fo.aggression.aggressive, "BLD_SHIPYARD_ORBITAL_DRYDOCK", 750, 5),
-    "BLD_SHIPYARD_CON_ADV_ENGINE": (0, "BLD_SHIPYARD_ORBITAL_DRYDOCK", 500, 5),
-    "BLD_SHIPYARD_AST": (fo.aggression.typical, "", 75, 5),
-    "BLD_SHIPYARD_AST_REF": (fo.aggression.maniacal, "BLD_SHIPYARD_AST", 500, 5),
-    "BLD_SHIPYARD_ORG_ORB_INC": (0, "BLD_SHIPYARD_BASE", 40, 8),
-    "BLD_SHIPYARD_ORG_CELL_GRO_CHAMB": (fo.aggression.aggressive, "BLD_SHIPYARD_ORG_ORB_INC", 64, 8),
-    "BLD_SHIPYARD_ORG_XENO_FAC": (fo.aggression.aggressive, "BLD_SHIPYARD_ORG_ORB_INC", 120, 8),
-    "BLD_SHIPYARD_ENRG_COMP": (fo.aggression.aggressive, "BLD_SHIPYARD_BASE", 200, 5),
-    "BLD_SHIPYARD_ENRG_SOLAR": (fo.aggression.maniacal, "BLD_SHIPYARD_ENRG_COMP", 1200, 5),
-    "BLD_NEUTRONIUM_FORGE": (fo.aggression.cautious, "BLD_SHIPYARD_BASE", 100, 3),
+    "BLD_SHIPYARD_BASE",
+    "BLD_SHIPYARD_ORBITAL_DRYDOCK",
+    "BLD_SHIPYARD_CON_NANOROBO",
+    "BLD_SHIPYARD_CON_GEOINT",
+    "BLD_SHIPYARD_CON_ADV_ENGINE",
+    "BLD_SHIPYARD_AST",
+    "BLD_SHIPYARD_AST_REF",
+    "BLD_SHIPYARD_ORG_ORB_INC",
+    "BLD_SHIPYARD_ORG_CELL_GRO_CHAMB",
+    "BLD_SHIPYARD_ORG_XENO_FAC",
+    "BLD_SHIPYARD_ENRG_COMP",
+    "BLD_SHIPYARD_ENRG_SOLAR",
+    "BLD_NEUTRONIUM_FORGE",
```

Comment:
In the long term, code from building_locations.py should be completely moved to buildings.py, then this set can also be moved there (and become a set or enums).

Reply:
I'd prefer to convert buildings to a module and move this to a separate file.

Reply:
Are all those buildings have ship hulls or ship parts which require those building?

Reply:
> Are all those buildings have ship hulls or ship parts which require those building? Maybe we can generate this list from FOCS data.

I guess we could, but we  still have to adapt the AI if the game gets more ship yards and the AI should learn to use it. I think it is not worth the effort. There'll always be a little code duplication between FOCS and AI.